Backpack/CardButton/Classes/ShareCardButton.swift

```diff
@@ -0,0 +1,147 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+public class BPKShareCardButton: UIButton {
```

Comment:
Is there any way we can have the share/save button inherit from a shared class? Right now it feels like we're having a duplicate implementation?

Reply:
This shared class can be private, so we still have a BPKShare and BPKSave button, but all the logic for look/feel would be in the super class.

Reply:
I will create a private BPKCardButton class for the duplicated functions : )

Reply:
I have moved the duplicate code to CardButtonProtocol's extension.

A parent class is not used because if the subclass is public, the parent class must be public as well.
I prefer not exposing the "abstract" class outside backpack. 

This is the commit. Let me know what you think
dcc8b9c3ec8d85fea61084f57e6fd95a39e99358